### PR TITLE
Fixes #4255 - Media details activity shows depictions in random languages, rather than in the user's locale

### LINKED ISSUE #4255: Media details activity shows depictions in random languages, rather than in the user's locale
**Steps to reproduce:** 

1. Find a picture that has depictions (there are many in Explore>Uploaded via mobile)
2. Open its media details
3. Finds its depictions
4. Most often they do not match your locale

**Commons app version:** 

Latest master 3.0.0-debug

**Screen-shots:** 

My locale is English, but depiction for [fire engine](https://www.wikidata.org/wiki/Q208281) is shown in [Luxembourgish](https://lb.wikipedia.org/wiki/Pompjeesgefier): "Pompjeesgefier" even though the Wikidata item has a label in English.

adb
https://commons.wikimedia.org/wiki/File:Fire_truck_in_Tokyo.jpg

The easiest way to investigate is probably to look at the web requests in logcat and see where Pompjeesgefier comes from.

### PR DESCRIPTION
**Pull request by Pratham2305**
**Description (required)**

Fixes #4255 

What changes did you make and why?

Since App fetches a list of a depiction in different languages but I have noticed that depiction is only shown in the first language given in list, which is causing this issue. I have added the conditions Now it will first check if the depiction caption is available in the user's locale and if not then it will also check for English as it is the default language, else show in any available language.

**Tests performed (required)**
Tested prodDebug on Pixel 3 with API level 29.

(image selected from the feature tab with depictions)
https://commons.m.wikimedia.org/wiki/File%3AD%C3%BClmen%2C_Merfeld%2C_Feldweg_am_M%C3%BChlenbach_--_2021_--_4278-80.jpg

default language: English

Latest Master 3.0.0-debug
>Screenshot_1614252218

Updated
> Screenshot_1614251879

### CODE REVIEW

**Comment by misaochan** [comment]
@Pratham2305 could you post a screenshot of the fixed media details view, please? Thanks!

**Comment by Pratham2305** [comment]
@misaochan Yes, Posted!

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -615,9 +616,14 @@ public void someResultsFound() {
      */
     private void buildDepictionList(List<IdAndCaptions> idAndCaptions) {
         depictionContainer.removeAllViews();
+        String locale = Locale.getDefault().getLanguage();
         for (IdAndCaptions idAndCaption : idAndCaptions) {
                 depictionContainer.addView(buildDepictLabel(
-                    idAndCaption.getCaptions().values().iterator().next(),
+                    //Check if the Depiction Caption is available in user's locale if not then check for english, else show any available.
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:622]: I personally love ternary operations but using 2 levels of ternary makes it hard to read like a very long sentence. I am good to merge this PR after you turned them to use if braces. :)

**Review by neslihanturan — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java (hunk 1 above)
Comment (Pratham2305) [reply app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java]: Done!

**Comment by misaochan** [comment]
Wonderful, thank you @Pratham2305 ! Merging since @neslihanturan 's review has been addressed. :)

**Pull request merged**